src/usbg_common_libconfig.c

```diff
@@ -187,3 +187,34 @@ int usbg_set_config_node_dev(config_setting_t *root,
 	return ret;
 }
 
+int usbg_set_config_node_guid(config_setting_t *root,
+				const char *node_name, void *val)
+{
+	const unsigned char *bin = *(char **)val;
+	const char xa[] = "0123456789abcdef";
+	config_setting_t *node;
+	char guid[GUID_CHAR_LENGTH + 1];
+	char *tmp = guid;
+	int ret = 0;
+
+	node = config_setting_add(root, node_name, CONFIG_TYPE_STRING);
+	if (!node)
+		return USBG_ERROR_NO_MEM;
+
+	/* convert binary representation to hex */
+	for (int i = 0; i < GUID_BIN_LENGTH; i++, *bin++) {
+
+		/* add the four dashes */
+		if ((i == 4) || (i == 6) || (i == 8) || (i == 10))
+			*tmp++ = '-';
+
+		*tmp++ = xa[((*bin) & 0xf0) >> 4];
+		*tmp++ = xa[((*bin) & 0x0f)];
+	}
+
+	sprintf(tmp, "\0");
```

Comment:
```suggestion
	snprintf(guid, GUID_CHAR_LENGTH + 1,
	         "%02x%02x%02x%02x-%02x%02x-%02x%02x-%02x%02x-%02x%02x%02x%02x%02x%02x",
	         bin[0], bin[1], bin[2], bin[3],
	         bin[4], bin[5], bin[6], bin[7],
	         bin[8], bin[9], bin[10], bin[11],
	         bin[12], bin[13], bin[14], bin[15]);
```

Reply:
Uugh, but sure, snprintf and sscanf is just ugly. Thanks for the changes! I will add them.